service/src/services/LibraryService.ts

```diff
@@ -124,15 +131,40 @@ export class LibraryService implements Service<fhir4.Library> {
     const query = extractIdentificationForQuery(args, params);
     const parsedParams = parseRequestSchema({ ...params, ...query }, LibraryDataRequirementsArgs);
 
+    // check to see if data requirements were already calculated for this Library and params
+    const dataReqsQuery: Filter<any> = {};
+    Object.entries(parsedParams).forEach(p => {
+      if (p[0] === 'id') {
+        dataReqsQuery.resourceId = p[1] as string;
+      } else {
+        dataReqsQuery[p[0]] = p[1] as string;
+      }
+    });
+
+    const dataReqs = await findDataRequirementsWithQuery(dataReqsQuery);
+
+    // if data requirements were already calculated for this Library and params, return them
+    if (dataReqs) {
+      logger.info('Successfully retrieved $data-requirements report');
```

Comment:
```suggestion
      logger.info('Successfully retrieved $data-requirements report from cache.');
```
Nitpick but adding some more context makes it easier to understand whether the data requirements were cached or if they are being calculated for the first time